### PR TITLE
feat: dark mode

### PR DESCRIPTION
**Pull request by diogogmatos**
Implemented a dark mode fully with TailwindCSS and the help of the `next-themes` npm package that handles toggling, saving preference to local storage, etc.

**Preview:**

image

image

image

### CODE REVIEW

**Comment by diogogmatos** [comment]
[feat: dark mode](https://trello.com/c/g56PwNCj/78-feat-dark-mode)

File: components/DarkModeToggler/DarkModeToggler.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Switch } from '@headlessui/react'
+import {useTheme} from 'next-themes';
```
Comment (ruilopesm) [inline components/DarkModeToggler/DarkModeToggler.tsx:2]:
```suggestion
import { useTheme } from 'next-themes';
```
Why is the formatter not catching this, btw?

File: components/DarkModeToggler/DarkModeToggler.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+function classNames(...classes) {
+    return classes.filter(Boolean).join(' ')
+  }
+
+const DarkModeToggler = () => {
+    const {theme, setTheme} = useTheme();
```
Comment (ruilopesm) [inline components/DarkModeToggler/DarkModeToggler.tsx:9]: Same here.

File: pages/_app.tsx [hunk 3]
```diff
@@ -1,8 +1,13 @@
 import type { AppProps } from "next/app";
 import "../styles/globals.css";
+import {ThemeProvider} from 'next-themes';
```
Comment (ruilopesm) [inline pages/_app.tsx:3]: Same here.

**Review by ruilopesm — APPROVED** [review]

**Pull request merged**